doc/HOWTO-KAFKA.md

```diff
@@ -0,0 +1,226 @@
+# Kafka exfiltration
+
+In the "daemon mode", Sonar stays memory-resident and pushes data to a network sink; one of these
+sinks is a Kafka broker.  See HOWTO-DAEMON.md for general information about this mode and the
+options available for configuring the Kafka producer in Sonar.
+
+Data and control messages are as described in HOWTO-DAEMON.md, with "topic", "key" and "value"
+having their standard Kafka meanings.
+
+## CONFIGURING A STANDARD KAFKA BROKER
+
+### Topics
+
+For each cluster with canonical name `<cluster>` that is going to be handled by the broker, the broker needs
+to be able to handle messages with these topics coming from the cluster:
+
+```
+<cluster>.sample
+<cluster>.sysinfo
+<cluster>.job
+<cluster>.cluster
+```
+
+The broker also needs to be able to handle these control topics (tentative, may change) that are
+sent from the back-end to the clients in the cluster:
+
+```
+<cluster>.control.node
+<cluster>.control.master
+```
+
+### Testing with Apache Kafka
+
+Test notes with standard Kafka server, see https://kafka.apache.org/quickstart.
+
+#### Setup
+
+You're going to be running several shells, let's call them Zookeeper, Server, Consumer, Work, and Sonar.
+
+The working directory for the following is the root directory of the unpacked Kafka distribution, eg
+`kafka_2.13-3.9.0/`.
+
+NOTE!  Currently Sonar does not work with Kafka 4.0.0, there's a protocol error, not debugged.  Use
```

Comment:
We might check if this is still true - that was a problem with kafka-rust, but librdkafka is probably better.  Either way, we might want to reword this.